app/src/organisms/ODD/QuickTransferFlow/utils/retrieveLiquidClassValues.ts

```diff
@@ -0,0 +1,369 @@
+import {
+  ETHANOL_LIQUID_CLASS_NAME,
+  getAllLiquidClassDefs,
+  getFlexNameConversion,
+  getLabwareDefURI,
+  GLYCEROL_LIQUID_CLASS_NAME,
+  linearInterpolate,
+  NONE_LIQUID_CLASS_NAME,
+  SAFE_MOVE_TO_WELL_OFFSET_FROM_TOP_MM,
+  WATER_LIQUID_CLASS_NAME,
+} from '@opentrons/shared-data'
+
+import { getTransferPlanAndReferenceVolumes } from './getTransferPlanAndReferenceVolumes'
+
+import type { LiquidHandlingPropertyByVolume } from '@opentrons/shared-data'
+import type { BlowOutLocation, QuickTransferSummaryState } from '../types'
+
+export const retrieveLiquidClassValues = (
+  state: QuickTransferSummaryState,
+  liquidHandlingAction: 'aspirate' | 'dispense' | 'all'
+): QuickTransferSummaryState => {
+  const { pipette } = state
+  const convertedPipetteName = getFlexNameConversion(pipette)
+
+  if (state.liquidClass.liquidClassName === NONE_LIQUID_CLASS_NAME) {
+    return getNoLiquidClassValues(
+      state,
+      convertedPipetteName,
+      liquidHandlingAction
+    )
+  } else {
+    return getLiquidClassValues(
+      state,
+      convertedPipetteName,
+      liquidHandlingAction
+    )
+  }
+}
+
+const convertBlowoutLocation = (
+  location: string | undefined,
+  state: QuickTransferSummaryState
+): BlowOutLocation | undefined => {
+  if (!location) return undefined
+
+  switch (location) {
+    case 'source':
+      return 'source_well'
+    case 'destination':
+      return 'dest_well'
+    case 'trash':
+      return state.dropTipLocation
+    default:
+      return undefined
+  }
+}
```

Comment:
not sure if this is a bit cleaner:

```suggestion
const convertBlowoutLocation = (
  state: QuickTransferSummaryState,
  location?: string
): BlowOutLocation | undefined => {
  switch (location) {
    case 'source':
      return 'source_well'
    case 'destination':
      return 'dest_well'
    case 'trash':
      return state.dropTipLocation
  }
}
```